### PR TITLE
Add TimeSeriesRegularizer and TimeSeriesImputer to the user guide

### LINKED ISSUE #3422: Update documentation to include TimeSeriesRegularizer and Imputer
Once #3376 and #3374 are merged in, we need a new section in the Time Series Modeling section of the User Guide to cover the new components!

### PR DESCRIPTION
**Pull request by eccabay**
Closes #3422

### CODE REVIEW

File: docs/source/user_guide/timeseries.ipynb [hunk 1]
```diff
@@ -89,6 +90,159 @@
...
+   "metadata": {},
+   "source": [
+    "### Time Series Regularizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the `TimeSeriesRegularizer` component to restore the missing and NaN DateTime values we've created in our data. This component is designed to infer the proper frequency using [Woodwork's `infer_frequency`](https://woodwork.alteryx.com/en/stable/generated/woodwork.statistics_utils.infer_frequency.html) function and generate a new DataFrame that follows it. In order to maintain as much original information from the input data as possible, all rows with completely correct times are ported over into this new DataFrame. If there are any rows that have the same timestamp as another, these will be dropped. The first occurrence of a data or time maintains priority. If there are any values that don\u2019t quite line up with the inferred frequency they will be shifted to any close missing times, or dropped if there are none nearby."
```
Comment (freddyaboulton) [inline docs/source/user_guide/timeseries.ipynb:165]: `The first occurence of a date`?

File: docs/source/user_guide/timeseries.ipynb [hunk 2]
```diff
@@ -89,6 +90,159 @@
...
+    "\n",
+    "regularizer = TimeSeriesRegularizer(time_index=\"Date\")\n",
+    "X_train, y_train = regularizer.fit_transform(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can see that pandas can once again successfully infer the frequency of the training data, and there are no more null values within `X_train`. However, by adding values that were dropped before, we have created NaN values within the target data, as well as the other columns in our training data."
```
Comment (freddyaboulton) [inline docs/source/user_guide/timeseries.ipynb:184]: `inferred the frequency`

File: docs/source/user_guide/timeseries.ipynb [hunk 3]
```diff
@@ -89,6 +90,159 @@
...
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fixing the data\n",
+    "\n",
+    "Sometimes, the datasets we work with do not have perfectly consistent DateTime columns. We can use the `TimeSeriesRegularizer` and `TimeSeriesImputer` to correct any disrepancies in our data in a time-series specific way.\n",
+    "\n",
+    "To show an example of this, let's create some discrepancies in our training data. We'll also add a couple of extra column in the X DataFrame to highlight more of the options with these tools."
```
Comment (freddyaboulton) [inline docs/source/user_guide/timeseries.ipynb:101]: `add a couple of extra columns`

File: docs/source/user_guide/timeseries.ipynb (hunk 1 above)
Comment (freddyaboulton) [inline docs/source/user_guide/timeseries.ipynb:165]: The formatting of the link got messed up: 
image

**Review by freddyaboulton — APPROVED** [review]
Thank you so much @eccabay ! Just left some minor nits.

File: docs/source/user_guide/timeseries.ipynb [hunk 4]
```diff
@@ -89,6 +90,159 @@
     "go.Figure(data=data, layout=layout)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fixing the data\n",
+    "\n",
+    "Sometimes, the datasets we work with do not have perfectly consistent DateTime columns. We can use the `TimeSeriesRegularizer` and `TimeSeriesImputer` to correct any disrepancies in our data in a time-series specific way.\n",
```
Comment (ParthivNaresh) [inline docs/source/user_guide/timeseries.ipynb:99]: Nit: *discrepancies

File: docs/source/user_guide/timeseries.ipynb [hunk 5]
```diff
@@ -89,6 +90,159 @@
...
+   "metadata": {},
+   "source": [
+    "### Time Series Regularizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the `TimeSeriesRegularizer` component to restore the missing and NaN DateTime values we've created in our data. This component is designed to infer the proper frequency using [Woodwork's \"infer_frequency\"](https://woodwork.alteryx.com/en/stable/generated/woodwork.statistics_utils.infer_frequency.html) function and generate a new DataFrame that follows it. In order to maintain as much original information from the input data as possible, all rows with completely correct times are ported over into this new DataFrame. If there are any rows that have the same timestamp as another, these will be dropped. The first occurrence of a date or time maintains priority. If there are any values that don\u2019t quite line up with the inferred frequency they will be shifted to any close missing times, or dropped if there are none nearby."
```
Comment (ParthivNaresh) [inline docs/source/user_guide/timeseries.ipynb:165]: Nit ***closely missing datetimes** in the last line

**Review by ParthivNaresh — APPROVED** [review]
Great work, just left a couple of nits!

**Pull request merged**